File: AD-HOC/1030/main.js
function survivor(n, k) {
    if (n == 1) {
        return 1;
    }
    return((survivor(n - 1, k) + k - 1) % n) + 1
}

let input = require('fs').readFileSync('/dev/stdin', 'utf-8');
let lines = input.split('\n');

nc = lines[0]
for (let i = 1; i <= nc; i++) {
    let [n, k] = lines[i].split(' ').map(Number);
    console.log(`Case ${i}: ${
        survivor(n, k)
    }`);
}

File: AD-HOC/1030/main.py
import sys
sys.setrecursionlimit(20000)


def survivor(n, k):
    if n == 1:
        return 1
    return ((survivor(n-1, k) + k - 1) % n) + 1


nc = int(input())
for i in range(1, nc+1):
    n, k = tuple(map(int, input().split()))
    print(f'Case {i}: {survivor(n, k)}')
